### PR TITLE
Go Azure Auth example

### PR DESCRIPTION
**Pull request by AnPucel**
# Description
A new Go code example w/ Azure auth

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New example (non-breaking change that adds a new code example)
- [ ] Other (Please describe what type of change)

# How Has This Been Tested?
This was tested on an Azure ubuntu VM w/ Vault running locally

### CODE REVIEW

File: go/7_auth-azure.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    if err != nil {
+      return metadataJson{}, fmt.Errorf("Error unmarshalling the response: %w", err)
+    }
+
+	return r, nil	
+}
+
+// Retrieves an access token from Azure MSI
+func getJWT() (string, error) {
+	// Create HTTP request for a managed services for Azure resources token to access Azure Resource Manager
```
Comment (digivava) [inline go/7_auth-azure.go:145]: Looks like there's an extra tab here causing the comment to indent

Comment (AnPucel) [reply go/7_auth-azure.go:145]: ooh. yeah. there's a couple of these. I'll address those. Thanks for the catch

File: go/7_auth-azure.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    var msi_endpoint *url.URL
+    msi_endpoint, err := url.Parse("http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01")
+    if err != nil {
+      return "", fmt.Errorf("Error creating URL: %w", err)
+
+    }
+
+	msi_parameters := msi_endpoint.Query()
```
Comment (digivava) [inline go/7_auth-azure.go:153]: Another extra tab here. Do you have the Go extension for VSCode installed? I feel like it should be fixing indentation issues on save

File: go/7_auth-azure.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      return "", fmt.Errorf("Error calling token endpoint: %w", err)
+    }
+
+    responseBytes,err := ioutil.ReadAll(resp.Body)
+    defer resp.Body.Close()
+    if err != nil {
+      return "", fmt.Errorf("Error reading response body: %w", err)
+    }
+
+    // Unmarshall response body into struct
```
Comment (digivava) [inline go/7_auth-azure.go:176]: In Go we spell unmarshal with one L ¯\_(ツ)_/¯

Comment (digivava) [reply go/7_auth-azure.go:176]: same in the error message bellow

File: go/7_auth-azure.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+		return "", fmt.Errorf("value type assertion failed: %T %#v", data[key], data[key])
+	}
+
+	return value, nil
+}
+
+// Retrieve instance metadata from Azure 
+func getMetadata() (metadataJson, error) {
+
+	var metadata_endpoint *url.URL
```
[the remark this answers is not in the document]
Comment (AnPucel) [reply go/7_auth-azure.go:105]: got it! Good to know. I got a little of this from Microsoft docs, so I'm a little surprised they wouldn't follow that.

File: go/7_auth-azure.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+// Retrieves an access token from Azure MSI
+func getJWT() (string, error) {
+	// Create HTTP request for a managed services for Azure resources token to access Azure Resource Manager
+    var msi_endpoint *url.URL
+    msi_endpoint, err := url.Parse("http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01")
+    if err != nil {
+      return "", fmt.Errorf("Error creating URL: %w", err)
+
```
Comment (digivava) [inline go/7_auth-azure.go:150]: extra empty line here

**Pull request merged**